ksql-api/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java

```diff
@@ -0,0 +1,201 @@
+  public void start(final Promise<Void> startPromise) {
+    httpServer = vertx.createHttpServer(httpServerOptions).requestHandler(setupRouter());
+    httpServer.exceptionHandler(ServerUtils::handleException);
+    Future<HttpServer> listenFuture = Promise.<HttpServer>promise().future();
+    Utils.connectPromise(listenFuture.map(s -> null), startPromise);
+    httpServer.listen(listenFuture);
+    vertx.getOrCreateContext().exceptionHandler(ServerUtils::handleException);
+  }
+
+  @Override
+  public void stop(final Promise<Void> stopPromise) {
+    if (httpServer == null) {
+      stopPromise.complete();
+    } else {
+      httpServer.close(stopPromise.future());
+    }
+  }
+
+  private Router setupRouter() {
+    Router router = Router.router(vertx);
+    router.route(HttpMethod.POST, "/query-stream").handler(BodyHandler.create())
+        .handler(this::handleQueryStream);
+    router.route(HttpMethod.POST, "/inserts-stream").handler(this::handleInsertsStream);
+    router.route(HttpMethod.POST, "/close-query").handler(BodyHandler.create())
+        .handler(this::handleCloseQuery);
+    return router;
+  }
+
+  private void handleInsertsStream(final RoutingContext routingContext) {
+    RecordParser recordParser = RecordParser
+        .newDelimited("\n", new InsertsBodyParser(endpoints, routingContext)::handleBodyBuffer);
+    routingContext.request()
+        .handler(recordParser);
+  }
+
+  private void handleQueryStream(final RoutingContext routingContext) {
+    HttpConnection conn = routingContext.request().connection();
+    ConnectionQueries connectionQueries = connectionsMap.get(conn);
+    if (connectionQueries == null) {
+      connectionQueries = new ConnectionQueries(conn);
+      connectionsMap.put(conn, connectionQueries);
+      conn.closeHandler(connectionQueries);
+      server.registerQueryConnection(conn);
+    }
+    JsonObject requestBody = routingContext.getBodyAsJson();
+    String sql = requestBody.getString("sql");
+    if (sql == null) {
+      handleError(routingContext.response(), 400, ERROR_CODE_MISSING_PARAM, "No sql in arguments");
+      return;
+    }
+    Boolean push = requestBody.getBoolean("push");
+    if (push == null) {
+      handleError(routingContext.response(), 400, ERROR_CODE_MISSING_PARAM, "No push in arguments");
+      return;
+    }
+    JsonObject properties = requestBody.getJsonObject("properties");
+    QueryPublisher queryPublisher = endpoints.createQueryPublisher(sql, push, properties);
+    QuerySubscriber querySubscriber = new QuerySubscriber(routingContext.response());
+
+    String queryID = server.registerQuery(querySubscriber);
+    connectionQueries.addQuery(queryID);
+    JsonObject metadata = new JsonObject();
+    metadata.put("column-names", queryPublisher.getColumnNames());
+    metadata.put("column-types", queryPublisher.getColumnTypes());
+    metadata.put("query-id", queryID);
+    if (!push) {
+      metadata.put("row-count", queryPublisher.getRowCount());
```

Comment:
Could we consider camelCase instead of kebab-case for the metadata object keys?

I have the JS client in mind, where it would be more natural to operate on objects with the former.
```js
const { columnNames, columnTypes } = queryMetadata;
// vs
const columnNames = queryMetadata["column-names"];
const columnTypes = queryMetadata["column-types"];
```

Reply:
+1 on camelCase - would be friendlier for JS.